Sources/TensorFlow/Loss.swift

```diff
@@ -65,6 +65,18 @@ public func hingeLoss<Scalar: TensorFlowFloatingPoint>(
     return max(Tensor(1) - expected * predicted, Tensor(0)).mean()
 }
 
+/// Returns the cosine proximity loss between predictions and expectations.
+///
+/// - Parameters:
+///   - predicted: Predicted outputs from a neural network.
+///   - expected: Expected values, i.e. targets, that correspond to the correct output.
+@differentiable(wrt: predicted)
+public func cosineProximityLoss<Scalar: TensorFlowFloatingPoint>(
```

Comment:
@rxwei wdyt?

Reply:
I'd suggest calling this `cosineSimilarity(_:_:)` and make it differentiable w.r.t. both parameters.

Rationale:
1. It is a standard measure of similarity and is the canonical name of this measure (it also has a [Wikipedia page](https://en.wikipedia.org/wiki/Cosine_similarity)).
2. It does not have to be a loss function, so it can be used in other contexts. Hence arguments labels differentiating between predicted and expected do not seem to be useful.
3. Because of (2), we should make it differentiable with respect to both parameters.

Reply:
I’ve just noticed that the argument labels were not removed as suggested. Could you please start a PR to resolve it when you get a chance? Thanks!

Reply:
@rxwei if it's okay I can fix this in #239

Reply:
Thanks for taking this on! It'd be better if you fix it in a separate CL instead.